### ISSUE TITLE
使用paddle.jit.load加载paddlehub与训练模型后计算梯度报错

### ISSUE THREAD

**Issue by ganyixian** [PaddlePaddle/Paddle#38690]
- 版本、环境信息：
    1）PaddlePaddle版本：paddlepaddle 1.8.5
    2）GPU：Cuda10.0
    3）系统环境：Linux Ubuntu 16.04.7; Python 2.7
- 模型信息
    1）模型名称: paddlehub pretrained yolov3_darknet53_vehicles
- 复现信息：
错误代码：
<img width="550" alt="截屏2022-01-04 下午5 37 17" src="https://user-images.githubusercontent.com/43914299/148039163-30a7443f-89ae-422e-b224-d96b230b2eb6.png">
报错信息：
<img width="749" alt="截屏2022-01-04 下午5 39 22" src="https://user-images.githubusercontent.com/43914299/148039441-6ce53295-d76b-435e-bfea-aa54a5b585da.png">

- 问题描述：使用paddle.jit.load加载paddlehub的预训练yolov3_darknet53_vehicles后执行预测，预测执行成功但是在使用backward() 方法计算梯度的时候报错：RunProgramGradOp 的对象域没有子域

**Comment by cuicheng01** [comment]
您好，这里使用可能有点问题，首先，inference model没有backward pass，其次，loss.backward()不应该放在model.eval()之后，另外注意到您使用的PaddlePaddle版本也比较低，建议以后使用PaddlePaddle最新的版本~

**Comment by ganyixian** [comment]
您好，可是我看到有教程是通过jit.load加载了通过save.inference.model接口保存的模型然后进行梯度计算的啊。如果模型没有backward pass的话请问有没有办法可以使用paddlehub的预训练模型进行预测并且计算梯度呢？谢谢！

**Comment by cuicheng01** [comment]
您好，方便贴一下您说的教程吗？您这边我理解是要进行预测，预测为什么要计算梯度呢？没太理解您的意思哈

**Comment by ganyixian** [comment]
paddle官网上jit.load的使用示例的第二部分
<img width="842" alt="截屏2022-01-05 下午3 22 34" src="https://user-images.githubusercontent.com/43914299/148176641-03a2ea65-76b2-4988-a8c2-09c44ed423fa.png">
<img width="823" alt="截屏2022-01-05 下午3 23 41" src="https://user-images.githubusercontent.com/43914299/148176782-5c2be1c3-9599-40d4-87df-95cc6583f796.png">
<img width="833" alt="截屏2022-01-05 下午3 23 56" src="https://user-images.githubusercontent.com/43914299/148176816-f6c174c1-eb33-4dda-9283-98bb5436210e.png">
<img width="836" alt="截屏2022-01-05 下午3 24 16" src="https://user-images.githubusercontent.com/43914299/148176851-1594d5a1-96dc-4b21-9c79-343b6244796c.png">

我是现在再尝试用paddle平台实现cw攻击算法，所以需要计算梯度，谢谢

**Comment by cuicheng01** [comment]
您好，这里是train的过程，是可以计算梯度的，您的代码是eval的过程，是不计算梯度的，改成train模式哈~

**Comment by ganyixian** [comment]
您好，那请问使用Paddlehub的与训练模型的时候有办法可以提取出中间层的结果吗？比如我现在在使用yolo_v3_darknet53的模型，我想提取出最后在nms之前的检测结果，除了使用静态图的program executor运行模型的program并抓去中间结果之外有别的办法吗？谢谢
